src/lib/holocene/maximizable/maximizable.svelte

```diff
@@ -0,0 +1,89 @@
+<script lang="ts">
+  import { onMount, type Snippet } from 'svelte';
+  import { twMerge as merge } from 'tailwind-merge';
+
+  import MaximizeButton from './button.svelte';
+
+  interface Props {
+    children: Snippet;
+    maximized: boolean;
+    onMaximize: (maximized: boolean) => void;
+    class?: string;
+    enabled?: boolean;
+    actions?: Snippet;
+  }
+
+  let {
+    children,
+    maximized,
+    onMaximize,
+    class: className = undefined,
+    enabled = true,
+    actions = undefined,
+  }: Props = $props();
+
+  let escapeListener = $state((event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      onMaximize(false);
+    }
+  });
+
+  $effect(() => {
+    if (maximized) {
+      document.addEventListener('keydown', escapeListener);
+    } else {
+      document.removeEventListener('keydown', escapeListener);
+    }
+  });
+
+  onMount(() => {
+    // on unmount
+    return () => {
+      document.removeEventListener('keydown', escapeListener);
+    };
+  });
+
+  const handleClick = () => {
+    onMaximize(!maximized);
+  };
+
+  const handleFocusOut = (event: FocusEvent) => {
+    if (
+      maximized &&
+      (!(event.currentTarget instanceof Element) ||
+        !(event.relatedTarget instanceof Element) ||
+        !event.currentTarget.contains(event.relatedTarget))
+    ) {
+      onMaximize(false);
+    }
+  };
+</script>
+
+<div
+  class={merge(
+    'relative',
+    maximized
+      ? 'fixed bottom-0 left-0 right-0 top-0 z-100 h-full w-full overflow-y-auto bg-white dark:bg-black'
+      : '',
```

Comment:
Do not use ternaries. Use the cslx stuff for this instead https://svelte.dev/docs/svelte/class#Attributes-Objects-and-arrays